src/Benchmarks/Program.cs

```diff
@@ -133,6 +140,21 @@ private static int GetThreadCount(IConfigurationRoot config)
             var threadCountValue = config["threadCount"];
             return threadCountValue == null ? -1 : int.Parse(threadCountValue);
         }
+
+        private static IConnectionFilter GetConnectionFilter(IConfigurationRoot config, IConnectionFilter prevFilter)
```

Comment:
It might be simpler to just handle all of the setting of `options.ConnectionFilter` in this method and remove the null-check + if in the calling code.

For instance this method will set `options.ConnectionFilter` to `null` when nothing is specified on the command line, but will daisy chain if a connection filter is set. This doesn't seem intended.